utils/s2n_random.c

```diff
@@ -442,6 +442,7 @@ S2N_RESULT s2n_rand_cleanup(void)
         ENGINE_finish(rand_engine);
         ENGINE_free(rand_engine);
         ENGINE_cleanup();
+        ENGINE_unregister_RAND(rand_engine);
```

Comment:
Shouldn't we be calling ENGINE_unregister_RAND before ENGINE_free?

I looked through the openssl code and ENGINE_unregister_RAND doesn't actually use rand_engine as anything other than a memory pointer, so it's unlikely to actually cause an error, but it's still a memory pointer we don't technically own anymore at this point, [since we freed it](https://github.com/openssl/openssl/blob/b5a635dc2113e1bc807ea358a670146c813df989/crypto/engine/eng_lib.c#L98).

Reply:
thanks - Will change.